token/program-2022/src/processor.rs

```diff
@@ -7602,4 +7677,249 @@ mod tests {
             )
         );
     }
+
+    #[test]
+    #[serial]
```

Comment:
nit: I see why you were running into the issue now -- `do_process_instruction_dups` doesn't set the syscalls. The best solution would be to have `do_process_instruction` call into `do_process_instruction_dups`, and set the syscalls once in `do_process_instruction_dups`, but we can do that in follow-up